transform/apply.go

```diff
@@ -36,6 +36,25 @@ const (
 	StatusSkipped = "skipped"
 )
 
+// Mode enumerates all the possible circumstances in which the transform
+// may be called
+type Mode string
+
+const (
+	// RMApply indicates the transform was executed as an "apply"
+	// meaning, the transform was run with no intension to save the
+	// output dataset
+	RMApply = Mode("apply")
+	// RMCommit indicates the transform was executed as a "commit"
+	// meaning, the transform was run with the intension to save the
+	// output dataset
+	RMCommit = Mode("commit")
+)
+
+func (m Mode) String() string {
+	return string(m)
+}
+
```

Comment:
Since we cannot import `transform` into the `event` package, added this helper func to convert `Modes` into `strings`.

However, having `Mode` be its own type may be overkill, based on how we are using it.

May be better to just have some const strings `ApplyMode = "apply"` & `CommitMode = "commit"`.

Reply:
I think your post-PR intuition is correct: we don't really need an extra type for the `RunMode` enumeration. Looking through this PR a few things jump out as a consequence of the extra type in the `transform` package:

* collection now needs to import transform
* seeing lots of `.String()` to-and-froming
* forces the `base` package to grow. We want to avoid having base be a catch-all dumping ground if we can

So, I think this comment is spot-on. Let's have `RMApply` and `RMCommit` be string constants `apply` and `commit` respectively